TalZivelin/TZivelin-HW-W01.js

```diff
@@ -0,0 +1,125 @@
+
+f) x = true - True, x is assigned true, so x is true.
+
+g) var x; x == 'undefined' - False, undefined and 'undefined' are not the same thing.
+
+h) '9'<'10' - False, Javascript reads the first number for comparison so 9 is not smaller than 1.
+
+i) typeof x + 1 === "number" - True if x is a number false if x is a string.
+
+j) typeof x % 2 === "number" - True is x is a number that results in a number, otherwise false.
+
+k) typeof (x % 2) === "number" - True, you either get a number, or NaN which returns typeof "Number".
+
+l) x++ == ++x - False because by the time you get to ++x the original x is one number larger.
+
+m) ++x == x++ - True, because x was increased and then compared to an increase in x.
+
+n) "1"+x == 1+x - When there is a plus things are converted to string, if the second x is number this would be false, if it's a string this will be true.
+
+o) "0"+1 == 1 - True, 01 and 1 are the same.
+
+p) (typeof (x+1))===(typeof x) - True because if x is a number it will be a number type on both sides, if it;s a string it will be a string type both places.
+
+q) (x*1 == x) || ((typeof x) != "number") - If x is a number you will get true you will get true otherwise this is false.
+
+r) (x=(typeof (x+(typeof x))))==x - True after you are doing the typeof calculation you are assigning it to x so yes x will equal x.
+
+All of the following can be solved with ordinary expressions and global variables with primitive values. You don't need functions, loops, or other topics beyond our first two classes.
+
+2) (Difficulty: easy)
+
+Assume variables x, y, and z are numbers.
+
+a) Write an expression for the mean (i.e. average) of x, y, and z. -  var mean = (x + y + z) / 3
+
+b) Write a series of expressions to adjust each of x, y, and z halfway toward their mean. That is, reset the value of each variable to something based on its previous value. 
+
+Use the mean variable from part a, 
+x = x + mean / 2; 
+y =y + mean / 2;
+z = z + mean / 2;
+	
+
+3) (Difficulty: easyish)
+
+Suppose you're encoding geometric shapes in a Cartesian (2D) coordinate system, and you represent a rectangle with four numeric variables:
+
+    l : horizontal position of left edge;
+    r : horizontal position of right edge;
+    t : vertical position of top edge;
+    b : vertical position of bottom edge.
+
+a) Write an expression for the rectangle's area. - a = (r - l) * (t - b)
+
+b) Write an expression which is true if the rectangle is taller than it is wide, and false otherwise. - (t - b) > (r - l)
+
+c) Write an expression for the circumference of the biggest circle which can fit inside the rectangle. (Hint: you'll need logic similar to that in b.) - 
+
+var width = (r - l);
+var height = (t - b);
+var diameter;
+ 
+if (width > height) 
+	diameter = Math.PI height;
+		else
+  	diameter = Math.PI width;
+
+
+d) Write an expression for the area of the smallest circle which completely encloses (i.e. circumscribes) the rectangle.
+
+r2 = width/2 + height/2
```

Comment:
3d) For this one, the way I tend to think about it is that we need the length of the diagonal from one corner of the rectangle to its opposite corner, since that would be the diameter of the smallest circle that can enclose it. In order to get that, we can go back to Pythagoras and a^2 + b^2 = c^2, using the two sides we already know as a and b.